swift-sdk/Internal/IterableAPIInternal.swift

```diff
@@ -383,20 +383,34 @@ final class IterableAPIInternal : NSObject, PushTrackerProtocol {
         }
     }
     
-    func updateSubscriptions(_ emailListIds: [String]?, unsubscribedChannelIds: [String]?, unsubscribedMessageTypeIds: [String]?) {
+    func updateSubscriptions(campaignId: NSNumber?, templateId: NSNumber?, emailListIds: [NSNumber]?, unsubscribedChannelIds: [NSNumber]?, unsubscribedMessageTypeIds: [NSNumber]?, subscribedMessageTypeIds: [NSNumber]?) {
```

Comment:
campaignId and templateId are in the API only for attribution and they're optional, they're not the main parameters for this call. If we decide to add them, they should be the last arguments in this method.

Reply:
+1. Also we should have default values so `campaignId: NSNumber? = nil` etc.